### DISCUSSION TITLE
Validate two optional fields which should only be validated together

### DISCUSSION THREAD

**Discussion by makks129** [go-playground/validator#1318] [Q&A]
I have a struct with 2 fields: `latitude` and `longitude`
Both of them are optional, but if 1 is present the other should be validated as required
I haven't found a combination that would work to validate these 4 cases:
```
1.
lat=nil
lng=nil
Valid
2.
lat=123
lng=nil
Error
3.
lat=nil
lng=123
Error
4.
lat=123
lng=123
Valid
```

I tried different variations of `required_` and `excluded_` and `omitempty` but one of the cases always failed so far
```go
Lat         string  `json:"lat" validate:"required_with=Inner.Lng,excluded_without=Inner.Lng,omitempty,latitude"`
Lng         string  `json:"lng" validate:"required_with=Inner.Lat,excluded_without=Inner.Lat,omitempty,longitude"`
```

It seems that it should be a fairly common case. Your help will be greatly appreciated!

**Comment by vynazevedo** [comment]
`required_with` on each field is enough here. Put it before `omitempty`, and refer to the sibling's Go field name:

```go
type Coordinates struct {
    Lat *float64 `json:"lat" validate:"required_with=Lng,omitempty,latitude"`
    Lng *float64 `json:"lng" validate:"required_with=Lat,omitempty,longitude"`
}
```

With `validate.Struct(...)`, that gives:

- Both nil: valid.
- Only one non-nil: the other fails `required_with`.
- Both present: both coordinate validators run.
- Both pointing to zero: valid; zero is a real coordinate.

The pointers distinguish a missing coordinate from zero. Your example declares strings, which can't be nil; if you keep strings, use the same tags and treat `""` as missing.

If these fields live inside `Inner`, the tags still say `Lng` and `Lat`: they're looked up relative to the struct containing the fields. You don't need `excluded_without` as well.

One detail in the examples: latitude `123` should fail the `latitude` check even when longitude is present. Use something like `45, 123` to test the valid pair.

Checked with validator v10.27.0, including a containing struct with an `Inner Coordinates` field. The [required_with documentation](https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Required_With) describes the presence check.
